xbmc/addons/VFSEntry.cpp

```diff
@@ -70,8 +70,13 @@ VFSEntryPtr CVFSAddonCache::GetAddonInstance(const std::string& strId, TYPE type
 
 void CVFSAddonCache::OnEvent(const AddonEvent& event)
 {
-  if (typeid(event) == typeid(AddonEvents::InstalledChanged))
+  if (typeid(event) == typeid(AddonEvents::Enabled) ||
+      typeid(event) == typeid(AddonEvents::Disabled) ||
+      typeid(event) == typeid(AddonEvents::ReInstalled) ||
+      typeid(event) == typeid(AddonEvents::UnInstalled))
+  {
     Update();
```

Comment:
Only maybe add a:
```cpp
  ADDON::AddonPtr addon;
  if (CServiceBroker::GetAddonMgr().GetAddon(enableEvent->id, addon, ADDON_VFS, false))
    Update();
```
to prevent not wanted update if e.g. YouTube addon is changed.

Reply:
Sure will add it.

Reply:
This won't work for event `UnInstalled` because `GetAddon()` will always return false. We need to update the cache in that case.